cluster-autoscaler/simulator/test_utils.go

```diff
@@ -0,0 +1,62 @@
+package simulator
+
+import (
+	"testing"
+
+	apiv1 "k8s.io/api/core/v1"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// InitializeClusterSnapshot clears cluster snapshot and then initializes it with given set of nodes and pods.
+// Both Spec.NodeName and Status.NominatedNodeName are used when simulating scheduling pods.
+func InitializeClusterSnapshot(
```

Comment:
Maybe InitializeClusterSnapshotOrDie and call t.Fail() on error? I think it's a pretty nice pattern for all those initializations in tests.

Reply:
> why do we need a NoError version
For sake of `TestFindNodesToRemove` which is not very sanely written. I will try to work around that and remove the NoError version